Content/Items/Starwood/Weapons.StarwoodScepter.cs

```diff
@@ -0,0 +1,741 @@
+			Item.rare = ItemRarityID.White;
+			Item.UseSound = SoundID.Item44;
+
+			Item.noMelee = true;
+			Item.DamageType = DamageClass.Summon;
+			Item.buffType = ModContent.BuffType<StarwoodSummonBuff>();
+			Item.shoot = ModContent.ProjectileType<StarwoodScepterSummonSplit>();
+		}
+
+		public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+		{
+			player.AddBuff(Item.buffType, 2);
+
+			Projectile proj = Projectile.NewProjectileDirect(source, Main.MouseWorld, Main.rand.NextVector2CircularEdge(5f, 5f), type, damage, knockback, player.whoAmI, ai2: 1f);
+			proj.originalDamage = Item.damage;
+
+			Projectile proj2 = Projectile.NewProjectileDirect(source, Main.MouseWorld, Main.rand.NextVector2CircularEdge(5f, 5f), type, damage, knockback, player.whoAmI, ai2: 0f);
+			proj2.originalDamage = Item.damage;
+
+			(proj.ModProjectile as StarwoodScepterSummonSplit).otherProj = proj2;
+			(proj2.ModProjectile as StarwoodScepterSummonSplit).otherProj = proj;
+
+			Helpers.DustHelper.DrawStar(Main.MouseWorld, ModContent.DustType<Dusts.GlowFastDecelerate>(), 5, 1, 1, 0.5f, 1, 0.5f, 0, -1, new Color(240, 200, 20));
+
+			return false;
+		}
+	}
+
+	public class StarwoodScepterSummonSplit : ModProjectile
+	{
+		public int lifetime;
+		public int empowermentTimer;
+		public int downtimeTimer;
+		public Projectile otherProj;
+		public Vector2 rotationalVelocity = Vector2.Zero;
+		public NPC MinionTarget
+		{
+			get
+			{
+				if (Owner.HasMinionAttackTargetNPC && Main.npc[Owner.MinionAttackTargetNPC].Distance(Projectile.Center) < 1000f)
+					return Main.npc[Owner.MinionAttackTargetNPC];
+
+				return null;
+			}
+		}
+
+		public Vector2 IdlePosition
+		{
+			get
+			{
+				if (IsParent)
+					return Owner.Center + new Vector2(-25 * Owner.direction, -50) + new Vector2(-35 * Owner.direction * Projectile.minionPos, 0f) + new Vector2(MathHelper.Lerp(5f, 15f, Utils.Clamp((float)Math.Sin(lifetime * 0.015f), 0, 1))).RotatedBy(MathHelper.ToRadians(lifetime));
+				else
+					return Owner.Center + new Vector2(-25 * Owner.direction, -50) + new Vector2(-35 * Owner.direction * otherProj.minionPos, 0f) + new Vector2(-MathHelper.Lerp(5f, 15f, Utils.Clamp((float)Math.Sin(lifetime * 0.015f), 0, 1))).RotatedBy(MathHelper.ToRadians(lifetime));
+			}
+		}
+
+		public StarwoodScepterSummonSplit OtherStar => otherProj.ModProjectile as StarwoodScepterSummonSplit;
+		public Projectile EmpoweredStar => Main.projectile.Where(p => p.active && p.type == ModContent.ProjectileType<StarwoodScepterSummonEmpowered>() && (p.ModProjectile as StarwoodScepterSummonEmpowered).Children.Contains(Projectile.whoAmI)).FirstOrDefault();
+		public bool HasEmpoweredStar => EmpoweredStar != null;
+		public bool IsEmpowered => Owner.GetModPlayer<StarlightPlayer>().empowered;
+		public bool FoundTarget => Target != null;
+		public ref float TargetWhoAmI => ref Projectile.ai[0];
```

Comment:
This is likely a poor candidate for an AI field, as whoAmI's change across clients

Reply:
wait even npc ones

Reply:
god terrarias netcode is so fucking atrocious